[opencv4] Build error with CUDA 12.2

Package: opencv4[contrib,core,cuda,cudnn,default-features,dnn,dnn-cuda,gtk,jpeg,png,quirc,tiff,webp]:x64-linux -> 4.8.0#5

**Host Environment**

- Host: x64-linux
- Compiler: GNU 11.4.0
-    vcpkg-tool version: 2023-09-15-ac02a9f660977426b8ec6392919fbb1d51b10998
    vcpkg-scripts version: 6a349fe1f 2023-09-18 (17 hours ago)

**To Reproduce**

`vcpkg install --recurse opencv4[dnn-cuda]`
**Failure logs**

```
-- Using cached opencv4-e5e1a3bfdea96bebda2ad963bc8f6cf17930aef7.patch.
-- Using cached opencv-opencv-4.8.0.tar.gz.
-- Cleaning sources at $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source $HOME/tmp/vcpkg/downloads/opencv-opencv-4.8.0.tar.gz
-- Applying patch 0001-disable-downloading.patch
-- Applying patch 0002-install-options.patch
-- Applying patch 0003-force-package-requirements.patch
-- Applying patch 0004-fix-eigen.patch
-- Applying patch 0005-fix-policy-CMP0057.patch
-- Applying patch 0006-fix-uwp.patch
-- Applying patch 0008-devendor-quirc.patch
-- Applying patch 0009-fix-protobuf.patch
-- Applying patch 0010-fix-uwp-tiff-imgcodecs.patch
-- Applying patch 0011-remove-python2.patch
-- Applying patch 0012-fix-zlib.patch
-- Applying patch 0015-fix-freetype.patch
-- Applying patch 0017-fix-flatbuffers.patch
-- Applying patch 0019-missing-include.patch
-- Applying patch $HOME/tmp/vcpkg/downloads/opencv4-e5e1a3bfdea96bebda2ad963bc8f6cf17930aef7.patch
-- Using source at $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean
-- Using cached opencv-cache/tiny_dnn/adb1c512e09ca2c7a6faef36f9c53e59-v1.0.0a3.tar.gz.
-- Using cached opencv-cache/nvidia_optical_flow/a73cd48b18dcc0cc8933b30796074191-edb50da3cf849840d680249aa6dbef248ebce2ca.zip.
-- Using cached opencv-opencv_contrib-4.8.0.tar.gz.
-- Cleaning sources at $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source $HOME/tmp/vcpkg/downloads/opencv-opencv_contrib-4.8.0.tar.gz
-- Applying patch 0007-fix-hdf5.patch
-- Applying patch 0016-fix-freetype-contrib.patch
-- Applying patch 0018-fix-depend-tesseract.patch
-- Applying patch 0019-fix-ogre-dependency.patch
-- Using source at $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean
-- Using cached opencv-cache/wechat_qrcode/238e2b2d6f3c18d6c3a30de0c31e23cf-detect.caffemodel.
-- Using cached opencv-cache/wechat_qrcode/cbfcd60361a73beb8c583eea7e8e6664-sr.caffemodel.
-- Using cached opencv-cache/wechat_qrcode/6fb4976b32695f9f5c6305c19f12537d-detect.prototxt.
-- Using cached opencv-cache/wechat_qrcode/69db99927a70df953b471daaba03fbef-sr.prototxt.
-- Using cached opencv-cache/xfeatures2d/boostdesc/0ea90e7a8f3f7876d450e4149c97c74f-boostdesc_bgm.i.
-- Using cached opencv-cache/xfeatures2d/boostdesc/232c966b13651bd0e46a1497b0852191-boostdesc_bgm_bi.i.
-- Using cached opencv-cache/xfeatures2d/boostdesc/324426a24fa56ad9c5b8e3e0b3e5303e-boostdesc_bgm_hd.i.
-- Using cached opencv-cache/xfeatures2d/boostdesc/202e1b3e9fec871b04da31f7f016679f-boostdesc_binboost_064.i.
-- Using cached opencv-cache/xfeatures2d/boostdesc/98ea99d399965c03d555cef3ea502a0b-boostdesc_binboost_128.i.
-- Using cached opencv-cache/xfeatures2d/boostdesc/e6dcfa9f647779eb1ce446a8d759b6ea-boostdesc_binboost_256.i.
-- Using cached opencv-cache/xfeatures2d/boostdesc/0ae0675534aa318d9668f2a179c2a052-boostdesc_lbgm.i.
-- Using cached opencv-cache/xfeatures2d/vgg/e8d0dcd54d1bcfdc29203d011a797179-vgg_generated_48.i.
-- Using cached opencv-cache/xfeatures2d/vgg/7126a5d9a8884ebca5aea5d63d677225-vgg_generated_64.i.
-- Using cached opencv-cache/xfeatures2d/vgg/7cd47228edec52b6d82f46511af325c5-vgg_generated_80.i.
-- Using cached opencv-cache/xfeatures2d/vgg/151805e03568c9f490a5e3a872777b75-vgg_generated_120.i.
-- Using cached opencv-cache/data/7505c44ca4eb54b4ab1e4777cb96ac05-face_landmark_model.dat.
-- Found external ninja('1.11.1').
-- Configuring x64-linux
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: $HOME/tmp/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake --build . --config Debug --target install -- -v -j17
    Working Directory: $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg
    See logs for more information:
      $HOME/tmp/vcpkg/buildtrees/opencv4/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/opencv4/portfile.cmake:466 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)



```
<details><summary>$HOME/tmp/vcpkg/buildtrees/opencv4/install-x64-linux-dbg-out.log</summary>

```
Change Dir: '$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg'

Run Build Command(s): $HOME/bin/ninja -v -v -j17 install
[1/1246] /usr/bin/c++ -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudev -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-undef -Wno-missing-declarations -Wno-unused-function -Wno-unused-variable -Wno-enum-compare -Wno-shadow -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/cudev/CMakeFiles/opencv_cudev.dir/src/stub.cpp.o -MF modules/cudev/CMakeFiles/opencv_cudev.dir/src/stub.cpp.o.d -o modules/cudev/CMakeFiles/opencv_cudev.dir/src/stub.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/src/stub.cpp
[2/1246] : && $HOME/tmp/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake -E rm -f lib/libopencv_cudev4d.a && /usr/bin/ar qc lib/libopencv_cudev4d.a  modules/cudev/CMakeFiles/opencv_cudev.dir/src/stub.cpp.o && /usr/bin/ranlib lib/libopencv_cudev4d.a && :
[3/1246] cd $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/core && $HOME/tmp/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake -DMODULE_NAME=core -DCL_DIR=$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/src/opencl -DOUTPUT=$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/core/opencl_kernels_core.cpp -P $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/cmake/cl2cpp.cmake
[4/1246] cd $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/core/CMakeFiles/cuda_compile_1.dir/src/cuda && $HOME/tmp/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake -E make_directory $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/core/CMakeFiles/cuda_compile_1.dir/src/cuda/. && $HOME/tmp/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=Debug -D generated_file:STRING=$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/core/CMakeFiles/cuda_compile_1.dir/src/cuda/./cuda_compile_1_generated_gpu_mat_nd.cu.o -D generated_cubin_file:STRING=$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/core/CMakeFiles/cuda_compile_1.dir/src/cuda/./cuda_compile_1_generated_gpu_mat_nd.cu.o.cubin.txt -P $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/core/CMakeFiles/cuda_compile_1.dir/src/cuda/cuda_compile_1_generated_gpu_mat_nd.cu.o.Debug.cmake
...
Skipped 527 lines
...
[502/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/lrn_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/lrn_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/lrn_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/lrn_layer.cpp
[503/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/normalize_bbox_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/normalize_bbox_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/normalize_bbox_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/normalize_bbox_layer.cpp
FAILED: modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/normalize_bbox_layer.cpp.o 
/usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/normalize_bbox_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/normalize_bbox_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/normalize_bbox_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/normalize_bbox_layer.cpp
In file included from $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/normalize_bbox_layer.cpp:50:
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/normalize_bbox.hpp: In instantiation of ‘void cv::dnn::cuda4dnn::NormalizeOp<T>::forward(const std::vector<cv::Ptr<cv::dnn::dnn4_v20230620::BackendWrapper> >&, const std::vector<cv::Ptr<cv::dnn::dnn4_v20230620::BackendWrapper> >&, cv::dnn::cuda4dnn::csl::Workspace&) [with T = __half]’:
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/normalize_bbox.hpp:89:14:   required from here
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/normalize_bbox.hpp:114:24: error: ambiguous overload for ‘operator!=’ (operand types are ‘__half’ and ‘double’)
  114 |             if (weight != 1.0)
      |                 ~~~~~~~^~~~~~
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/normalize_bbox.hpp:114:24: note: candidate: ‘operator!=(int, double)’ (built-in)
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/normalize_bbox.hpp:114:24: note: candidate: ‘operator!=(long long unsigned int, double)’ (built-in)
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/normalize_bbox.hpp:114:24: note: candidate: ‘operator!=(long long int, double)’ (built-in)
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/normalize_bbox.hpp:114:24: note: candidate: ‘operator!=(long unsigned int, double)’ (built-in)
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/normalize_bbox.hpp:114:24: note: candidate: ‘operator!=(long int, double)’ (built-in)
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/normalize_bbox.hpp:114:24: note: candidate: ‘operator!=(unsigned int, double)’ (built-in)
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/normalize_bbox.hpp:114:24: note: candidate: ‘operator!=(float, double)’ (built-in)
In file included from /usr/local/cuda/include/cuda_fp16.h:4070,
                 from /usr/local/cuda/include/cublas_api.h:77,
                 from /usr/local/cuda/include/cublas_v2.h:69,
                 from $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/csl/cublas.hpp:14,
                 from $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../op_cuda.hpp:11,
                 from $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/normalize_bbox_layer.cpp:45:
/usr/local/cuda/include/cuda_fp16.hpp:714:42: note: candidate: ‘bool operator!=(const __half&, const __half&)’
  714 | __CUDA_HOSTDEVICE__ __forceinline__ bool operator!=(const __half &lh, const __half &rh) { return __hneu(lh, rh); }
      |                                          ^~~~~~~~
[504/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/fully_connected_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/fully_connected_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/fully_connected_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/fully_connected_layer.cpp
[505/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/mvn_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/mvn_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/mvn_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/mvn_layer.cpp
[506/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/padding_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/padding_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/padding_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/padding_layer.cpp
[507/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/proposal_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/proposal_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/proposal_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/proposal_layer.cpp
[508/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/permute_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/permute_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/permute_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/permute_layer.cpp
[509/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/region_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/region_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/region_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/region_layer.cpp
FAILED: modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/region_layer.cpp.o 
/usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/region_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/region_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/region_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/region_layer.cpp
In file included from $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/region_layer.cpp:58:
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/region.hpp: In instantiation of ‘void cv::dnn::cuda4dnn::RegionOp<T>::forward(const std::vector<cv::Ptr<cv::dnn::dnn4_v20230620::BackendWrapper> >&, const std::vector<cv::Ptr<cv::dnn::dnn4_v20230620::BackendWrapper> >&, cv::dnn::cuda4dnn::csl::Workspace&) [with T = __half]’:
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/region.hpp:94:14:   required from here
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/region.hpp:124:35: error: ambiguous overload for ‘operator>’ (operand types are ‘__half’ and ‘int’)
  124 |             if (nms_iou_threshold > 0) {
      |                 ~~~~~~~~~~~~~~~~~~^~~
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/region.hpp:124:35: note: candidate: ‘operator>(int, int)’ (built-in)
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/region.hpp:124:35: note: candidate: ‘operator>(long long unsigned int, int)’ (built-in)
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/region.hpp:124:35: note: candidate: ‘operator>(long long int, int)’ (built-in)
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/region.hpp:124:35: note: candidate: ‘operator>(long unsigned int, int)’ (built-in)
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/region.hpp:124:35: note: candidate: ‘operator>(long int, int)’ (built-in)
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/region.hpp:124:35: note: candidate: ‘operator>(unsigned int, int)’ (built-in)
$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/primitives/region.hpp:124:35: note: candidate: ‘operator>(float, int)’ (built-in)
In file included from /usr/local/cuda/include/cuda_fp16.h:4070,
                 from /usr/local/cuda/include/cublas_api.h:77,
                 from /usr/local/cuda/include/cublas_v2.h:69,
                 from $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../cuda4dnn/csl/cublas.hpp:14,
                 from $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/../op_cuda.hpp:11,
                 from $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/region_layer.cpp:44:
/usr/local/cuda/include/cuda_fp16.hpp:720:42: note: candidate: ‘bool operator>(const __half&, const __half&)’
  720 | __CUDA_HOSTDEVICE__ __forceinline__ bool operator> (const __half &lh, const __half &rh) { return __hgt(lh, rh); }
      |                                          ^~~~~~~~
[510/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/prior_box_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/prior_box_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/prior_box_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/prior_box_layer.cpp
[511/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/nary_eltwise_layers.cpp
[512/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/reduce_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/reduce_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/reduce_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/reduce_layer.cpp
[513/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/reorg_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/reorg_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/reorg_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/reorg_layer.cpp
[514/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/elementwise_layers.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/elementwise_layers.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/elementwise_layers.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/elementwise_layers.cpp
[515/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/pooling_layer.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/pooling_layer.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/pooling_layer.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/pooling_layer.cpp
[516/1246] /usr/bin/c++ -DCV_CUDA4DNN=1 -DCV_OCL4DNN=1 -DENABLE_PLUGINS -DHAVE_FLATBUFFERS=1 -DHAVE_PROTOBUF=1 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/dnn -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean/modules/cudev/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/core/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/imgproc/include -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/misc/tflite -I$HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/3rdparty/include/opencl/1.2 -isystem $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg -isystem $HOME/tmp/vcpkg/installed/x64-linux/include -isystem /usr/local/cuda/include -fPIC   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-undef -Wno-invalid-offsetof -g  -O0 -DDEBUG -D_DEBUG -std=c++11 -fPIC -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/recurrent_layers.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/recurrent_layers.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/recurrent_layers.cpp.o -c $HOME/tmp/vcpkg/buildtrees/opencv4/src/4.8.0-8224011a81.clean/modules/dnn/src/layers/recurrent_layers.cpp
[517/1246] cd $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudaarithm/CMakeFiles/cuda_compile_1.dir/src/cuda && $HOME/tmp/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake -E make_directory $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudaarithm/CMakeFiles/cuda_compile_1.dir/src/cuda/. && $HOME/tmp/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=Debug -D generated_file:STRING=$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudaarithm/CMakeFiles/cuda_compile_1.dir/src/cuda/./cuda_compile_1_generated_reduce.cu.o -D generated_cubin_file:STRING=$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudaarithm/CMakeFiles/cuda_compile_1.dir/src/cuda/./cuda_compile_1_generated_reduce.cu.o.cubin.txt -P $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudaarithm/CMakeFiles/cuda_compile_1.dir/src/cuda/cuda_compile_1_generated_reduce.cu.o.Debug.cmake
[518/1246] cd $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudawarping/CMakeFiles/cuda_compile_1.dir/src/cuda && $HOME/tmp/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake -E make_directory $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudawarping/CMakeFiles/cuda_compile_1.dir/src/cuda/. && $HOME/tmp/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=Debug -D generated_file:STRING=$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudawarping/CMakeFiles/cuda_compile_1.dir/src/cuda/./cuda_compile_1_generated_remap.cu.o -D generated_cubin_file:STRING=$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudawarping/CMakeFiles/cuda_compile_1.dir/src/cuda/./cuda_compile_1_generated_remap.cu.o.cubin.txt -P $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudawarping/CMakeFiles/cuda_compile_1.dir/src/cuda/cuda_compile_1_generated_remap.cu.o.Debug.cmake
[519/1246] cd $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudawarping/CMakeFiles/cuda_compile_1.dir/src/cuda && $HOME/tmp/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake -E make_directory $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudawarping/CMakeFiles/cuda_compile_1.dir/src/cuda/. && $HOME/tmp/vcpkg/downloads/tools/cmake-3.27.1-linux/cmake-3.27.1-linux-x86_64/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=Debug -D generated_file:STRING=$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudawarping/CMakeFiles/cuda_compile_1.dir/src/cuda/./cuda_compile_1_generated_warp.cu.o -D generated_cubin_file:STRING=$HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudawarping/CMakeFiles/cuda_compile_1.dir/src/cuda/./cuda_compile_1_generated_warp.cu.o.cubin.txt -P $HOME/tmp/vcpkg/buildtrees/opencv4/x64-linux-dbg/modules/cudawarping/CMakeFiles/cuda_compile_1.dir/src/cuda/cuda_compile_1_generated_warp.cu.o.Debug.cmake
ninja: build stopped: subcommand failed.
```
</details>

This issue has been hinted at already in https://github.com/microsoft/vcpkg/issues/32673#issuecomment-1646084287.
It is known to opencv (opencv/opencv/issues/23893), has been fixed there (opencv/opencv/issues/24104, not tested by me), but this has not been released yet.

--- a/ports/opencv4/0020-fix-compat-cuda12.2.patch
+++ b/ports/opencv4/0020-fix-compat-cuda12.2.patch
@@ -1,0 +1,32 @@
+commit ab8cb6f8a9034da2a289b84685c6d959266029be
+Author: cudawarped <12133430+cudawarped@users.noreply.github.com>
+Date:   Tue Aug 1 13:02:42 2023 +0300
+
+    cuda: fix for compatibility with CUDA Toolkit >= 12.2.0
+
+diff --git a/modules/dnn/src/cuda4dnn/primitives/normalize_bbox.hpp b/modules/dnn/src/cuda4dnn/primitives/normalize_bbox.hpp
+index f067dddaa7..91ff33f817 100644
+--- a/modules/dnn/src/cuda4dnn/primitives/normalize_bbox.hpp
++++ b/modules/dnn/src/cuda4dnn/primitives/normalize_bbox.hpp
+@@ -111,7 +111,7 @@ namespace cv { namespace dnn { namespace cuda4dnn {
+              * or there might be several weights
+              * or we don't have to scale
+              */
+-            if (weight != 1.0)
++            if (weight != static_cast<T>(1.0f))
+             {
+                 kernels::scale1_with_bias1<T>(stream, output, input, weight, 1.0);
+             }
+diff --git a/modules/dnn/src/cuda4dnn/primitives/region.hpp b/modules/dnn/src/cuda4dnn/primitives/region.hpp
+index d22d44214e..3af05155fe 100644
+--- a/modules/dnn/src/cuda4dnn/primitives/region.hpp
++++ b/modules/dnn/src/cuda4dnn/primitives/region.hpp
+@@ -121,7 +121,7 @@ namespace cv { namespace dnn { namespace cuda4dnn {
+                 new_coords
+             );
+ 
+-            if (nms_iou_threshold > 0) {
++            if (nms_iou_threshold > static_cast<T>(0.0f)) {
+                 auto output_mat = output_wrapper->getMutableHostMat();
+                 CV_Assert(output_mat.type() == CV_32F);
+                 for (int i = 0; i < input.get_axis_size(0); i++) {

--- a/ports/opencv4/portfile.cmake
+++ b/ports/opencv4/portfile.cmake
@@ -28,6 +28,7 @@ vcpkg_from_github(
       0015-fix-freetype.patch
       0017-fix-flatbuffers.patch
       0019-missing-include.patch
+      0020-fix-compat-cuda12.2.patch
       "${ARM64_WINDOWS_FIX}"
 )
 # Disallow accidental build of vendored copies

--- a/ports/opencv4/vcpkg.json
+++ b/ports/opencv4/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "opencv4",
   "version": "4.8.0",
-  "port-version": 6,
+  "port-version": 7,
   "description": "computer vision library",
   "homepage": "https://github.com/opencv/opencv",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6082,7 +6082,7 @@
     },
     "opencv4": {
       "baseline": "4.8.0",
-      "port-version": 6
+      "port-version": 7
     },
     "opendnp3": {
       "baseline": "3.1.1",

--- a/versions/o-/opencv4.json
+++ b/versions/o-/opencv4.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "ef78c1958b122045e9d1e353150049431b3162fa",
+      "version": "4.8.0",
+      "port-version": 7
+    },
+    {
       "git-tree": "6a1280a0a3854032ba4ae9172b509ce46f81795f",
       "version": "4.8.0",
       "port-version": 6

